f8a_report/sentry_report_helper.py

```diff
@@ -0,0 +1,132 @@
+                                headers={"Authorization": auth})
+            # Check for status code
+            # If status is not success, log it as an error
+            if resp.status_code == 200:
+                logger.info('Successfully invoked Sentry API \n {resp}'.format(resp=resp.json()))
+                # associate the retrieved data to result
+                result = self.normalize_sentry_data(start_date, end_date, resp.json())
+            else:
+                logger.error('Error received from SSentry API \n {resp}'.format(resp=resp.json()))
+        except Exception as e:
+            logger.error('Unable to invoke Sentry API. Reason: %r' % e)
+        return result
+
+    def normalize_sentry_data(self, start_date, end_date, errorlogs):
+        """Retrieve results for selected worker from RDB."""
+        report_type = 'sentry-error-data'
+        report_name = dt.strptime(end_date, '%Y-%m-%d').strftime('%Y-%m-%d')
+        result = {
+            "error_report": {}
+        }
+        # Iterating all the error logs
+        for item in errorlogs:
+            errors = {}
+            events = self.retrieve_events(item['id'])
+            errors['id'] = item['id']
+            errors['lastSeen'] = item['lastSeen']
+            errors[events['pods_impacted']] = item['metadata']['type'] + ": " + \
+                item['metadata']['value'] if item['metadata'].get('type')\
+                else item['metadata']['title']
+            errors['stacktrace'] = events['stacktrace']
+            # Detecting the endpoint services
+            endpointpodname = events['pods_impacted'].split('-')
+            endpointpodname.pop(len(endpointpodname) - 1)
+            endpointpodname.pop(len(endpointpodname) - 1)
+            server_name = "-".join(endpointpodname)
+            result['error_report'][server_name] = result['error_report'][server_name] \
+                if result['error_report'].get(server_name) else {}
+            # Calculating total errors
+            result['error_report'][server_name]['total_errors'] = \
+                result['error_report'][server_name]['total_errors'] + 1 \
+                if result['error_report'][server_name].get('total_errors') else 1
+            if not result['error_report'][server_name].get('errors'):
+                result['error_report'][server_name]['errors'] = []
+            result['error_report'][server_name]['errors'].append(errors)
+            # Saving the final report in the relevant S3 bucket
+
+            try:
+                obj_key = '{depl_prefix}/{type}/{report_name}.json'.format(
+                    depl_prefix=self.s3.deployment_prefix, type=report_type, report_name=report_name
+                )
+                self.s3.store_json_content(content=result, obj_key=obj_key,
+                                           bucket_name=self.s3.report_bucket_name)
+            except Exception as e:
```

Comment:
Have some specific exceptions handled here with more meaningful exception statements for each of this exception.

Reply:
as discussed @miteshvp , this will be handled in later stages